src/header/common/authorization.rs

```diff
@@ -22,14 +22,14 @@ impl<S: Scheme> DerefMut for Authorization<S> {
     }
 }
 
-impl<S: Scheme> Header for Authorization<S> {
+impl<S: Scheme + 'static> Header for Authorization<S> where <S as FromStr>::Err: 'static {
```

Comment:
Huh, seems odd to need this bound, but I see why. Not something I would immediately think of...